staging/src/k8s.io/apiserver/pkg/storage/feature/feature_support_checker.go

```diff
@@ -79,29 +81,58 @@ func (f *defaultFeatureSupportChecker) Supports(feature storage.Feature) (bool,
 
 // CheckClient accepts client and calculate the support per endpoint and caches it.
 // It will return at any point if error happens or one endpoint is not supported.
-func (f *defaultFeatureSupportChecker) CheckClient(ctx context.Context, c client, feature storage.Feature) error {
+func (f *defaultFeatureSupportChecker) CheckClient(ctx context.Context, cancel context.CancelFunc, c client, feature storage.Feature) error {
 	switch feature {
 	case storage.RequestWatchProgress:
-		return f.clientSupportsRequestWatchProgress(ctx, c)
+		return f.checkClient(ctx, cancel, c)
 	default:
 		return fmt.Errorf("feature %q is not implemented in DefaultFeatureSupportChecker", feature)
 
 	}
 }
 
-func (f *defaultFeatureSupportChecker) clientSupportsRequestWatchProgress(ctx context.Context, c client) error {
+func (f *defaultFeatureSupportChecker) checkClient(ctx context.Context, cancel context.CancelFunc, c client) error {
+	go func(ctx context.Context) {
+		// start with 10 ms, multiply by 2 each step, 16 steps ~= 10 minutes
+		backoff := wait.Backoff{
+			Duration: 10 * time.Millisecond,
+			Jitter:   0,
+			Factor:   2,
+			Steps:    16,
+		}
+		err := wait.ExponentialBackoffWithContext(ctx, backoff, func(ctx context.Context) (done bool, err error) {
+			for _, ep := range c.Endpoints() {
+				err := f.clientSupportsRequestWatchProgress(ctx, c, ep)
+				// Iterate until we get err == nil
+				if err != nil {
+					return false, nil
+				}
+			}
+			supports, _ := DefaultFeatureSupportChecker.Supports(storage.RequestWatchProgress)
+			if !supports {
+				klog.InfoS("RequestWatchProgress feature is not supported by provided etcd. This will prevent ConsistentListFromCache from working")
+			}
+			return true, nil
+		})
+		if err != nil {
+			klog.ErrorS(err, "Failed to check client support after retrying")
+		}
+		cancel()
+	}(ctx)
+	return nil
```

Comment:
```suggestion
	
	// start with 10 ms, multiply by 2 each step, 16 steps ~= 10 minutes
	backoff := wait.Backoff{
			  Duration: 10 * time.Millisecond,
			  Jitter:   0,
			  Factor:   2,
			  Steps:    16,
	 }
	f.lock.Lock()
	defer f.lock.Unlock()
	for _, ep := range c.Endpoints() {
	  if found := f.checkingEndpoint[ep]; found {
	    continue
	  }
	  f.checkingEndpoint[ep] = struct{}{}
	  go func(ep string) {
		  err := wait.ExponentialBackoffWithContext(ctx, backoff, func(ctx context.Context) (done bool, err error) {
		     err := f.clientSupportsRequestWatchProgress(ctx, c, ep)
		     return err != nil, nil
		  })
		supports, _ := f.Supports(storage.RequestWatchProgress)
		if !supports {
			klog.InfoS("RequestWatchProgress feature is not supported by provided etcd. This will prevent ConsistentListFromCache from working")
		}
		if err != nil {
			  klog.ErrorS(err, "Failed to check client support after retrying")
		}
	  }(ep)
	}
	return nil
```